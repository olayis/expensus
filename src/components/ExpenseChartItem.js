import React from 'react';
import { Chart } from 'react-google-charts';
import spinner from '../../public/img/spinner.gif';

const ExpenseChartItem = (props) => (
  <div className='expense-chart'>
    <div className='expense-chart__title'>{props.title}</div>
    <Chart
      className='expense-chart__chart'
      width={props.width}
      height={props.height}
      chartType={props.chartType}
      loader={
        <div className='expense-chart__loader'>
          <img src={spinner} alt='' width='200' height='200' />
          <p>Loading Chart...</p>
        </div>
      }
      data={props.data}
      rootProps={props.chartRootProps}
      options={props.options}
    />
  </div>
);

export default ExpenseChartItem;
