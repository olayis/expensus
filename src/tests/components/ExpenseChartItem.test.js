import React from 'react';
import { shallow } from 'enzyme';
import ExpenseChartItem from '../../components/ExpenseChartItem';

test('should render ExpenseChartItem with chart data', () => {
  const wrapper = shallow(
    <ExpenseChartItem
      title='Chart Title'
      width='100%'
      height='50rem'
      chartType='Chart Type'
      data={[1, 2, 3]}
      chartRootProps={{ 'data-testid': '1' }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
