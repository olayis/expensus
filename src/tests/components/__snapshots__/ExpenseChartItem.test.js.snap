// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseChartItem with chart data 1`] = `
<div
  className="expense-chart"
>
  <div
    className="expense-chart__title"
  >
    Chart Title
  </div>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="Chart Type"
    chartWrapperParams={Object {}}
    className="expense-chart__chart"
    columns={null}
    controls={null}
    data={
      Array [
        1,
        2,
        3,
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="50rem"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div
        className="expense-chart__loader"
      >
        <img
          alt=""
          height="200"
          src="test-file-stub"
          width="200"
        />
        <p>
          Loading Chart...
        </p>
      </div>
    }
    options={
      Object {
        "colors": null,
      }
    }
    render={null}
    rootProps={
      Object {
        "data-testid": "1",
      }
    }
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="100%"
  />
</div>
`;
